infra/Pos/Network/Windows/defaultdns.c

```diff
@@ -0,0 +1,75 @@
+#include <winsock2.h>
+#include <iphlpapi.h>
+#include <stdio.h>
+#include <string.h>
+#include <windows.h>
+
+#define MALLOC(x) HeapAlloc(GetProcessHeap(), 0, (x))
+#define FREE(x) HeapFree(GetProcessHeap(), 0, (x))
+
+char *getWindowsDefDnsServer(void) {
+    FIXED_INFO *pFixedInfo;
+    ULONG ulOutBufLen;
+    DWORD dwRetVal;
+
+// This function isn't thread-safe because of this 'static' declaration.
+// Care is required when calling it.
+    static char dnsIPAddress[16] = "";
+    char allocationErr[] = "Error allocating memory needed to call GetNetworkParams\n";
+
+    pFixedInfo = (FIXED_INFO *) MALLOC(sizeof (FIXED_INFO));
+        if (pFixedInfo == NULL) {
+            printf(allocationErr);
+            return dnsIPAddress;
+        }
+        ulOutBufLen = sizeof (FIXED_INFO);
+
+// Make an initial call to GetAdaptersInfo to get the necessary size into the
+// ulOutBufLen variable
+    if (GetNetworkParams(pFixedInfo, &ulOutBufLen) == ERROR_BUFFER_OVERFLOW) {
+        FREE(pFixedInfo);
+        pFixedInfo = (FIXED_INFO *) MALLOC(ulOutBufLen);
+        if (pFixedInfo == NULL) {
+            printf(allocationErr);
+            return dnsIPAddress;
+        }
+    }
+
+    switch (dwRetVal = GetNetworkParams(pFixedInfo, &ulOutBufLen)) {
+    case ERROR_BUFFER_OVERFLOW:
```

Comment:
Lets simplify this error handling. We don't actually care that much about all the different failure cases. Lets follow the pattern from https://msdn.microsoft.com/en-us/library/windows/desktop/aa365968(v=vs.85).aspx and just check for success or not-success. e.g.

    if (dwRetVal = GetNetworkParams(pFixedInfo, &ulOutBufLen) == NO_ERROR) {

and in the failure case return NULL, only return the `dnsIPAddress` in the success case.